lib/java/ydoc-server/src/main/java/org/enso/ydoc/Main.java

```diff
@@ -19,36 +13,11 @@ public static void main(String[] args) throws Exception {
         "helidon.serialFilter.pattern",
         "javax.management.**;java.lang.**;java.rmi.**;javax.security.auth.Subject;!*");
 
-    var ydoc = Main.class.getResource(YDOC_SERVER_PATH);
-    var contextBuilder = WebEnvironment.createContext().allowIO(IOAccess.ALL);
-
     Sampling.init();
 
-    // Can't use try-with-resource in ExecutorService because API was added in JDK19
-    var executor = Executors.newSingleThreadExecutor();
-    try {
-      try (var parser = new ParserPolyfill()) {
-        var ydocJs =
-            Source.newBuilder("js", ydoc).mimeType("application/javascript+module").build();
-
-        CompletableFuture.supplyAsync(contextBuilder::build, executor)
-            .thenAcceptAsync(
-                ctx -> {
-                  WebEnvironment.initialize(ctx, executor);
-                  parser.initialize(ctx);
-                  ctx.eval(ydocJs);
-                },
-                executor)
-            .get();
-      }
-      System.out.println("Press enter to exit");
-      System.in.read();
-    } finally {
-      executor.shutdown();
-      var terminated = executor.awaitTermination(10, TimeUnit.SECONDS);
-      if (!terminated) {
-        executor.shutdownNow();
-      }
+    try (var ydoc = new Ydoc.Builder().build()) {
+      ydoc.start();
+      lock.acquire();
```

Comment:
Is it intentional that this Semaphore is never released?

Reply:
yes, we are waiting for any interruption signal here. Otherwise, the main process ends.